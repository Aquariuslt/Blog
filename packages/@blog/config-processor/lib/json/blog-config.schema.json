{
  "$schema": "http://json-schema.org/draft-07/schema",
  "$id": "http://github.com/aquariuslt/blog/config.schema.json",
  "title": "Blog Configuration",
  "description": "blog config file yml specification",
  "type": "object",
  "properties": {
    "version": {
      "type": "string",
      "default": "1.0.0"
    },
    "site": {
      "$ref": "#/definitions/site"
    },
    "build": {
      "$ref": "#/definitions/build"
    },
    "features": {
      "$ref": "#/definitions/features"
    }
  },
  "required": ["version", "site", "build"],
  "definitions": {
    "site": {
      "type": "object",
      "properties": {
        "host": {
          "type": "string",
          "format": "url",
          "description": "Your host, e.g. http://your-site.com or https://my-site.com."
        },
        "title": {
          "type": "string",
          "description": "Your blog title."
        },
        "author": {
          "type": "string",
          "description": "Your name"
        },
        "avatar": {
          "type": "string",
          "description": "Your profile picture"
        },
        "subtitle": {
          "type": "string",
          "description": "Your blog sub title"
        },
        "description": {
          "type": "string",
          "description": "Your blog description, this usually will built in html head meta"
        },
        "language": {
          "type": "string",
          "description": "Your prefer language",
          "enum": ["zh", "en"],
          "default": "zh"
        },
        "copyright": {
          "type": "string",
          "description": "Site copyright, usually appear under your website",
          "default": "copyright 2019 aquariuslt blog"
        }
      },
      "required": ["host", "title", "subtitle", "description"]
    },
    "build": {
      "type": "object",
      "properties": {
        "directory": {
          "type": "object",
          "properties": {
            "posts": {
              "type": "string",
              "description": "markdown posts file base directory, relative to project root path",
              "default": "posts"
            },
            "public": {
              "type": "string",
              "description": "build location path",
              "default": "dist"
            }
          },
          "required": []
        },
        "favicon": {
          "type": "string",
          "description": "Your favicon file location",
          "default": "favicon.ico"
        },
        "theme": {
          "type": "string",
          "description": "Your blog theme package name, not implementation now",
          "default": "vue"
        },
        "colors": {
          "type": "object",
          "description": "Your blog theme color preset from material color customization",
          "default": {}
        }
      },
      "required": []
    },
    "features": {
      "type": "object",
      "properties": {
        "ga": {
          "type": "string",
          "description": "Your Google Analytics tracking id"
        },
        "disqus": {
          "type": "string",
          "description": "Your disqus short id"
        }
      },
      "required": []
    }
  }
}
