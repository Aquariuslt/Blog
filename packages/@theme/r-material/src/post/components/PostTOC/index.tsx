import * as React from 'react';

import { makeStyles, StyleRules, Theme, useTheme } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';

import styles from '../../../styles';

const useStyles = makeStyles((theme: Theme): StyleRules => ({
    root: {
      width: styles.sidebar.width,
      flexShrink: 0,
      order: 2,
      position: 'sticky',
      overflowX: 'hidden',
      overflowY: 'auto',
      overflowWrap: 'break-word',
      padding: theme.spacing(2, 2, 2, 0),
      display: 'none',
      [theme.breakpoints.up('md')]: {
        display: 'block'
      },
      listStyleType: 'none'
    },
    contents: {
      marginTop: theme.spacing(2),
      paddingLeft: theme.spacing(1.5)
    },
    ul: {
      padding: 0,
      margin: 0,
      listStyleType: 'none'
    },
    item: {
      fontSize: 13,
      padding: theme.spacing(0.5, 0, 0.5, 1),
      borderLeft: '4px solid transparent',
      boxSizing: 'content-box',
      '&:hover': {
        borderLeft: `4px solid ${theme.palette.type === 'light' ? theme.palette.grey[200] : theme.palette.grey[900]}`,
        cursor: 'pointer'
      },
      '&$active,&:active': {
        borderLeft: `4px solid ${theme.palette.type === 'light' ? theme.palette.grey[300] : theme.palette.grey[800]}`
      }
    }
  })
);

interface PostTOCProps {
  contents: BlogModel.ContentItem[] | undefined;
}


const PostTOC: React.ComponentType<PostTOCProps> = (props: PostTOCProps): JSX.Element => {

  const theme = useTheme();
  const classes = useStyles();


  const ContentLink = (item: BlogModel.ContentItem): JSX.Element => {
    return (
      <React.Fragment>
        <Typography
          component="li"
          className={classes.item}
          style={{
            paddingLeft: theme.spacing((item.level))
          }}
        >
          {item.label}
        </Typography>
        <Typography component="ul" className={classes.ul}>
          {
            item.children.length > 0 &&
            item.children.map((child): JSX.Element => (
              <ContentLink key={child.id}{...child} />
            ))
          }
        </Typography>
      </React.Fragment>
    );
  };

  return (
    <nav className={classes.root}>
      {
        props.contents &&
        props.contents.map((item): JSX.Element => (
          <ContentLink key={item.id} {...item}/>
        ))
      }
    </nav>
  );
};


export default PostTOC;
