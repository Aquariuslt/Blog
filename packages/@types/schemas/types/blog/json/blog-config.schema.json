{
  "$schema": "http://json-schema.org/draft-07/schema",
  "$id": "http://github.com/aquariuslt/blog/config.schema.json",
  "description": "blog config file yml specification",
  "type": "object",
  "properties": {
    "version": {
      "type": "string",
      "default": "1.0.0"
    },
    "site": {
      "$ref": "#/definitions/site"
    },
    "build": {
      "$ref": "#/definitions/build"
    }
  },
  "required": [
    "version",
    "site",
    "build"
  ],
  "definitions": {
    "site": {
      "type": "object",
      "properties": {
        "host": {
          "type": "string",
          "format": "url"
        },
        "title": {
          "type": "string"
        },
        "subtitle": {
          "type": "string"
        },
        "description": {
          "type": "string"
        },
        "language": {
          "type": "string",
          "enum": [
            "zh",
            "en"
          ],
          "default": "zh"
        },
        "copyright": {
          "type": "string",
          "default": "copyright 2019 aquariuslt blog"
        }
      },
      "required": [
        "host",
        "title",
        "subtitle",
        "description"
      ]
    },
    "build": {
      "type": "object",
      "properties": {
        "directory": {
          "type": "object",
          "properties": {
            "posts": {
              "type": "string",
              "description": "markdown posts file base directory, relative to project root path",
              "default": "posts"
            },
            "public": {
              "type": "string",
              "description": "build location path",
              "default": "dist"
            }
          },
          "required": []
        },
        "favicon": {
          "type": "string",
          "default": "favicon.ico"
        },
        "theme": {
          "type": "string",
          "default": "vue"
        }
      },
      "required": []
    }
  }
}
