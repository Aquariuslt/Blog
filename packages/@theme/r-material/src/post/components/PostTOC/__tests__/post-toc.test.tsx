import * as _ from 'lodash';
import * as React from 'react';
import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';

import PostTOC from '../';

import * as mockTOC from './__mocks__/sample-toc.json';

Enzyme.configure({ adapter: new Adapter() });

describe('@theme/r-material: post/component: PostTOC', (): void => {
  it('# should mount correctly', (): void => {
    const wrapper = Enzyme.shallow(<PostTOC contents={mockTOC} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('# should mount w/o contents correctly', (): void => {
    const wrapper = Enzyme.shallow(<PostTOC contents={undefined} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('# should handle click toc label to trigger scoll animation', () => {
    const wrapper = Enzyme.mount(<PostTOC contents={mockTOC} />);

    const lastItem = _.last(mockTOC);
    const id = _.get(lastItem, 'id') || '';

    // since hash item in PostContent, we create a dom element to mock
    let el = document.createElement('div');
    el.setAttribute('id', id);
    document.body.appendChild(el);

    // since window.scrollTo not impl in jsdom, before we use puppeteer as env, we stub window.scrollTo function
    window.scrollTo = jest.fn();

    wrapper
      .find(`[id="${id}"]`)
      .childAt(0)
      .simulate('click');

    expect(wrapper).toMatchSnapshot();
  });
});
