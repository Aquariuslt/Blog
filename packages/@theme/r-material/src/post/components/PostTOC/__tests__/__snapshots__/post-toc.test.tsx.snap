// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@theme/r-material: post/component: PostTOC # should mount correctly 1`] = `
<nav
  className="makeStyles-root-1"
>
  <ContentLink
    id="google-codejam-2016-qualification"
    key="google-codejam-2016-qualification"
    label="Google CodeJam 2016 Qualification"
    level={1}
    pid={-1}
    position={0}
  >
    <Component />
    <Component />
    <Component />
  </ContentLink>
</nav>
`;

exports[`@theme/r-material: post/component: PostTOC # should mount w/o contents correctly 1`] = `
<nav
  className="makeStyles-root-1"
/>
`;
